### DISCUSSION TITLE
Sorting files by name

### DISCUSSION THREAD

**Discussion by Angelk90** [community/community#139025] [Repositories]
### Select Topic Area

General

### Body

I was looking at this project: https://github.com/zed-industries/zed

But I don't understand how the sorting of files works on github.
First come the elements with the dot, then the elements with the uppercase letters and finally the elements with all uppercase characters?

<img width="142" alt="Screenshot 2024-09-18 alle 20 01 08" src="https://github.com/user-attachments/assets/80c03980-3c51-47a8-9d9f-c603c3584f83">

Here is a screenshot of how vscode sorts the elements, the elements are sorted correctly.

<img width="214" alt="Screenshot 2024-09-18 alle 19 55 30" src="https://github.com/user-attachments/assets/7613b40b-ce3a-487b-aa2e-6962beb81023">

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @Angelk90 !

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the `Repositories` category. I've gone ahead and moved it for you. Good luck!

**Reply by Angelk90** [reply]
is there anything new?

**Comment by diepei** [comment]
Hi there!

What you see here is about different file sorting algorithms between GitHub and VS Code. 

GitHub uses case-sensitive sorting (where uppercase letters come before lowercase ones) while **VS Code uses case-insensitive sorting** (ignoring whether letters are uppercase or lowercase when comparing them)

Thank you!!! 🫡

**Reply by Angelk90** [reply]
@diepei : I see, which order do you prefer?
Which one do you think is more correct?

**Reply by diepei** [reply]
> @diepei : I see, which order do you prefer?
> 
> Which one do you think is more correct?

In my opinion case-insensitive sorting (vscode) is generally considered more user-friendly and is the default in most modern operating systems and file managers because:
1. It makes files easier to find since users don't need to remember whether something started with an uppercase or lowercase letter
2. It matches how humans naturally think about alphabetical ordering

I prefer the case-insensitive sorting
